### ISSUE TITLE
Could move the FWords: TWordsCollection; to be a protected member of TSkLabel? 

### ISSUE THREAD

**Issue by wqmeng** [skia4delphi/skia4delphi#370]
Could move the FWords: TWordsCollection; to be a protected member of TSkLabel? 

Recently I would like to add the onMouseOver, onMouseLeave, onMouseEnter event to the TCustomWordsItem object, and I extend the TCustomWordsItem, And also create a  TSkLabelEx class to extend the TSkLabel class. 

Then override the function of the TSkLabel class in TSkLabelEx to support the worditem event the mouseOver, MouseEnter, MouseLeave. Now the only matter is the FWords: TWordsCollection which is a private member and can not accessed by my sub-class,  So the first word item will not be changed to be my TCustomWordsItemEx object which can not handle the mouseEvent.

Thank you.

```
constructor TSkLabelEx.Create(AOwner: TComponent);
begin
  inherited;
  FWords.Free;
  FWords := TWordsCollection.Create(Self, TCustomWordsItemEx);
//  Words.set
  FWords.OnChange := WordsChange;
end;
```

The FWords can not access,  and if changed to Words.Free, Words := TWordsCollection.Create(Self, TCustomWordsItemEx);  The FWords.Assign function will fail too as the FWords are already freed.  If do not free the old FWords, the first WordItem will always be TCustomWordsItem  instead of **TCustomWordsItemEx** .

I want to change the default FWords to be a TWordsCollection of my sub-class TCustomWordsItemEx which add the mouse Event support. 

Thank you.

**Comment by viniciusfbb** [comment]
@wqmeng First, thank you very much for the report, I think this change is very valid.

However, after the integration with RAD Studio, we followed Embarcadero's guidelines, that is, the changes above the implementation are very limited so as not to break the binary compatibility of the DCUs between the minor versions of RAD Studio. In other words, it is only possible to implement these changes for RAD Studio 13.0.

**Comment by TommiPrami** [comment]
Yo,

Just a food for a thought.

Could you open up a branch for RAD studio 13, where you could start to implement these breaking changes.

And if someone starts using it, it would be their own risk, or burden if there is a rocky road ahead of them.

Just my 0.02€

-tee-

**Comment by viniciusfbb** [comment]
> Could you open up a branch for RAD studio 13, where you could start to implement these breaking changes.

@TommiPrami Good point! We already have it: [v7-lts branch](https://github.com/skia4delphi/skia4delphi/tree/v7-lts) with some minor changes for RAD Studio 13.

**Comment by viniciusfbb** [comment]
@wqmeng You can also:
1. Clone skia4delphi source;
2. Make your changes;
3. Install inno setup;
4. Open and run skia4delphi\Tools\Setup\Setup.iss;

I usually prefer to avoid manual steps...

**Comment by viniciusfbb** [comment]
For archaeology: The PR #391 is the related to this discussion.

**Comment by viniciusfbb** [comment]
> find all the Skia.Package.RTL290.bpl and Skia.Package.VCL290.bpl ,Skia.Package.FMX290.bpl in your system, and delete them.

@wqmeng Ahh. You're right about that. From RAD Studio 12.0 onwards, Skia4Delphi is already installed in the IDE, and the bpls that are in the `$(BDS)\bin` folder are always automatically loaded by the IDE and have priority over any other, so for the IDE to use the bpls generated by the setup at design-time, you would actually have to delete the Skia bpls that are in the `$(BDS)\bin` folder. This is only necessary for changes at design-time (in the IDE), since at runtime, the linker prioritizes the dcp based only on the Library Path order, so the dcp generated by the setup will be prioritized.
